src/charm.py

```diff
@@ -95,6 +100,13 @@ def _dex_auth_layer(self) -> Layer:
         }
         return Layer(layer_config)
 
+    @property
+    def _issuer_url(self) -> str:
+        """Return issuer-url value if config option exists; otherwise default Dex's endpoint."""
+        if issuer_url := self.model.config["issuer-url"]:
+            return issuer_url
+        return f"http://{self._app_name}.{self._namespace}.svc:5556/dex"
```

Comment:
Should we use `dex_port`?

Reply:
The suggestion is good. To correctly configure Dex's http port we have to pass the `http_port` arg to Dex's command, but nowhere in the code are we doing this. I think that to avoid burying that change in this PR, I will open an issue and change it in a different one. Does that make sense?